axelrod/strategies/adaptivezerodet.py

```diff
@@ -0,0 +1,120 @@
+from axelrod.action import Action
+from axelrod.player import Player
+from axelrod.random_ import random_choice
+from typing import Tuple
+from typing import List
+
+C, D = Action.C, Action.D
+
+class AdaptiveZeroDet(Player):
+	name = 'AdaptiveZeroDet'
+	classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': True,
+        'makes_use_of': set(["game"]),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+	def __init__(self, phi: float = 0.125, s: float = 0.5, l: float = 3, four_vector: Tuple[float, float, float, float] = None, initial: Action = C) -> None:
+		# This Keeps track of the parameter values (phi,s,l) as well as the four vector which makes final decisions.
+		self.scores = {C: 0, D: 0}
+		self.phi = phi
+		self.s = s
+		self.l = l

```

Comment:
Could you use some more meaningful variable names instead of 's', 'l' and 'd', please?